py/server/tests/test_table_data_service.py

```diff
@@ -357,9 +360,17 @@ def test_partition_size_sub_failure(self):
         data_service = TableDataService(backend)
         backend.sub_partition_size_fail_test = True
         table = data_service.make_table(TableKeyImpl("test"), refreshing=True)
+        table.coalesce()
+        # wait for the size subscription failure callback to be called for 5 seconds
+        for _ in range(50):
+            time.sleep(0.1)
+            if backend.is_size_sub_failure_cb_called:
+                break
+        else:
+            self.fail("size subscription failure callback was not called in 5s")
```

Comment:
This is probably better than what you had, although I really don't like seeing this kind of sleep loop; does Python not have condition variables?